### DISCUSSION TITLE
Picking n-th spectra from a mgf file

### DISCUSSION THREAD

**Discussion by hsogaard** [levitsky/pyteomics#103] [Q&A] [answered]
Hello,

I would like to read a `.mgf` file with thousands of spectra in it, and then pick every n-th spectra in the file, to create a smaller subset of the data in a new `.mgf` file. I want to use `mgf.write()` in a for loop so that I look through the file, apply the desired logic, and then write the spectra that passes the "filter" into the new subset file.  

I can read the file perfectly fine, and I can easily print every n-th spectra by indexing like this: `mgfFile[n]`, in a for loop. However, I encounter some issues when trying to write these results to a `.mgf` file.

For example, taking every tenth spectra, doing something along these lines:

    file = mgf.read(filePath)
    nSpectra = len(file)
    
    for i in range(nSpectra):
        if i % 10 == 0:
            mgf.write(file[i], outputFile)
Or by using `with`.

In other words, something like this:

    open and read mgf file: 
        for each spectrum in file:
            if spectrumIndex % 10 == 0:
                mgf.write(spectrum[with matching index], outputFile)

Any help is much appreciated! 
Thanks!

**Comment by mobiusklein** [comment, marked as answer]
`mgf.write` takes an `Iterable` over spectra, not a single spectrum as its first argument. Without seeing an error message, I have to guess this is the problem you've encountered. You can rewrite this two ways to make it work:

```python
file = mgf.read(filePath)
nSpectra = len(file)

for i in range(nSpectra):
    if i % 10 == 0:
        mgf.write([file[i]], outputFile)
```
Or
```python
file = mgf.read(filePath)
nSpectra = len(file)
mgf.write((file[i] for i in range(nSpectra) if i % 10), outputFile)
```
The first solves the problem by calling `write`  and passing a list of spectra of length 1 for every spectrum you want to write.
The second solves the problem by converting your for-loop to a generator expression and passing the resulting iterator object to `write`

**Reply by hsogaard** [reply]
Thank you! I was missing the extra pair of brackets in the `write` function. The addition of these did the trick!
